examples/sklearn-linear/job_configs/sklearn_linear_base/app/custom/linear_learner.py

```diff
@@ -0,0 +1,110 @@
+        self.random_state = random_state
+        self.train_data = None
+        self.valid_data = None
+        self.n_samples = None
+        self.local_model = None
+
+    def initialize(self, fl_ctx: FLContext):
+        self.fl_ctx = fl_ctx
+        data = self.load_data()
+        self.train_data = data["train"]
+        self.valid_data = data["valid"]
+        # train data size, to be used for setting
+        # NUM_STEPS_CURRENT_ROUND for potential aggregation
+        self.n_samples = data["train"][-1]
+        self.n_features = data["train"][0].shape[1]
+        # model will be created after receiving gobal parameters
+
+    def set_parameters(self, params):
+        self.local_model.coef_ = params["coef"]
+        if self.local_model.fit_intercept:
+            self.local_model.intercept_ = params["intercept"]
+
+    def train(self, curr_round: int, global_param: Optional[dict] = None) -> dict:
+        (x_train, y_train, train_size) = self.train_data
+        if curr_round == 0:
+            # initialize model with global_param
+            # and set to all zero
+            fit_intercept = bool(global_param["fit_intercept"])
+            self.local_model = SGDClassifier(
+                loss=global_param["loss"],
+                penalty=global_param["penalty"],
+                fit_intercept=fit_intercept,
+                learning_rate=global_param["learning_rate"],
+                eta0=global_param["eta0"],
+                max_iter=1,
+                warm_start=True,
+                random_state=self.random_state,
+            )
+            n_classes = global_param["n_classes"]  # Binary classification
+            self.local_model.classes_ = np.array([i for i in range(n_classes)])
+            self.local_model.coef_ = np.zeros((1, self.n_features))
+            if fit_intercept:
+                self.local_model.intercept_ = np.zeros((1,))
+        # Training starting from global model
+        # Note that the parameter update using global model has been performed
+        # during global model evaluation
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore")
+            self.local_model.fit(x_train, y_train)
+        if self.local_model.fit_intercept:
+            params = {
+                "coef": self.local_model.coef_,
+                "intercept": self.local_model.intercept_,
+            }
+        else:
+            params = {"coef": self.local_model.coef_}
+        return copy.deepcopy(params), self.local_model
+
+    def evaluate(self, curr_round: int, global_param: Optional[dict] = None) -> dict:
+        # set local model with global parameters
+        self.set_parameters(global_param)
+        # perform validation
+        (x_valid, y_valid, valid_size) = self.valid_data
+        y_pred = self.local_model.predict(x_valid)
+        auc = roc_auc_score(y_valid, y_pred)
+        metrics = {"AUC": auc}
+        return metrics, self.local_model
```

Comment:
this is returning a tuple of (metrics, self.local_model)
is this expected?

Reply:
yes the executor expect it https://github.com/NVIDIA/NVFlare/blob/dev/nvflare/app_opt/sklearn/sklearn_executor.py

Reply:
Same here for the return type hint